app/src/main/java/com/example/sharewhatyoucanproject/rdashboard/rposts/RPostViewModel.kt

```diff
@@ -0,0 +1,64 @@
+package com.example.sharewhatyoucanproject.rposts
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.example.sharewhatyoucanproject.models.GeoPoint
+import com.example.sharewhatyoucanproject.models.PostModel
+import com.google.android.gms.tasks.OnCompleteListener
+import com.google.firebase.firestore.FirebaseFirestore
+import kotlin.collections.ArrayList
+
+class RPostViewModel(
+    private val db: FirebaseFirestore,
+) : ViewModel() {
+    var arrayList: ArrayList<PostModel> = mutableListOf<PostModel>() as ArrayList<PostModel>
+    private val _taskResult = MutableLiveData<TaskResult>()
+    val taskResult: LiveData<TaskResult> = _taskResult
+
+    fun getlist() {
+        db.collection("posts")
```

Comment:
will you integrate paging in a future PR? Does Firebase handle paging automatically?

Reply:
Sorry I didn't setup yet, I will integrate in future PR's.